### PR TITLE
authoring/tools-for-developers: Fix link inside a note 

### PR DESCRIPTION
**Pull request by xmnlab**
Add tools for development to menu and fix link inside the note.

I just saw that the other filenames use underscore instead of dash. 
I created the file using dash, should I change to underscore? In general dash is more used or recommend (for example https://support.google.com/webmasters/answer/76329?hl=en), but I can change to underscore with no problem.

note: menu was already added! thanks @lwasser

### CODE REVIEW

**Comment by lwasser** [comment]
@xmnlab yes!! underscores are not as good for SEO as dashes are. so i'm actually going to be renaming most files in this repo. THANK YOU so much for this.

**Comment by xmnlab** [comment]
thanks @lwasser .. I can wait your PRs to add the menu item according to you change.

**Comment by choldgraf** [comment]
+1 for dashes everywhere :-)

**Comment by lwasser** [comment]
@xmnlab yay update from master and you'll see my start to changing things -- dashes galore! i haven't changed it everywhere knowing several other PR's (like this one) are open. but i will work towards that over the next few weeks!! also note the new repo name :)

**Comment by choldgraf** [comment]
I think that the one thing we should make sure of is that we don't make the file names so complicated that they are hard to parse by humans, even if they're better-parsed by the machines :-)

**Comment by xmnlab** [comment]
@lwasser I rebased the branch on top of upstream/master :)

File: authoring/tools-for-developers.md [hunk 1]
```diff
@@ -64,8 +64,7 @@ repos:
 ```
 
 ```{note}
-See [the flake8 hooks explanation](see: https://flake8.pycqa.org/en/latest/user/using-hooks.html) 
-for more details
+See <a href="https://flake8.pycqa.org/en/latest/user/using-hooks.html">the flake8 hooks explanation</a> for more details.
```
Comment (choldgraf) [inline authoring/tools-for-developers.md:67]: why are we hard-coding HTML here? In general we should try to use markdown syntax...

Comment (xmnlab) [reply authoring/tools-for-developers.md]: oh sorry, actually I copied/pasted the code you sent in the review and as it didn't work I thought it was some conflict with `note` section. I just fixed the typo and it worked. thanks! as it was the first time I am working with jupyter-book ... I was a little bit confused ... thanks for the review !

**Comment by xmnlab** [comment]
rebased again!

**Comment by xmnlab** [comment]
@lwasser thanks for adding authoring/tools-for-developers !
I just kept the fixing for the link inside a note. thanks!
